src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,529 @@
+                    {
+                        return false;
+                    }
+                    if (!TryParseProcessorArchitecture(attributeValue, out ProcessorArchitecture arch))
+                    {
+                        return false;
+                    }
+                    flags |= (AssemblyNameFlags)(((int)arch) << 4);
+                }
+                else if (IsAttribute(attributeName, "Retargetable"))
+                {
+                    if (!TryRecordNewSeen(ref alreadySeen, AttributeKind.Retargetable))
+                    {
+                        return false;
+                    }
+
+                    if (attributeValue.Equals("Yes", StringComparison.OrdinalIgnoreCase))
+                    {
+                        flags |= AssemblyNameFlags.Retargetable;
+                    }
+                    else if (attributeValue.Equals("No", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // nothing to do
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+                else if (IsAttribute(attributeName, "ContentType"))
+                {
+                    if (!TryRecordNewSeen(ref alreadySeen, AttributeKind.ContentType))
+                    {
+                        return false;
+                    }
+
+                    if (attributeValue.Equals("WindowsRuntime", StringComparison.OrdinalIgnoreCase))
+                    {
+                        flags |= (AssemblyNameFlags)(((int)AssemblyContentType.WindowsRuntime) << 9);
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    // Desktop compat: If we got here, the attribute name is unknown to us. Ignore it.
+                }
+
+                if (!TryGetNextToken(out _, out token))
+                {
+                    return false;
+                }
+            }
+
+            result = new AssemblyNameParts(name, version, cultureName, flags, pkt);
+            return true;
+        }
+
+        private static bool IsAttribute(string candidate, string attributeKind)
+            => candidate.Equals(attributeKind, StringComparison.OrdinalIgnoreCase);
+
+        private static bool TryParseVersion(string attributeValue, ref Version? version)
+        {
+#if NET8_0_OR_GREATER
+            ReadOnlySpan<char> attributeValueSpan = attributeValue;
+            Span<Range> parts = stackalloc Range[5];
+            parts = parts.Slice(0, attributeValueSpan.Split(parts, '.'));
+#else
+            string[] parts = attributeValue.Split('.');
+#endif
+            if (parts.Length is < 2 or > 4)
+            {
+                return false;
+            }
+
+            Span<ushort> versionNumbers = stackalloc ushort[4];
```

Comment:
IMO the stackalloc pattern here unnecessarily introduces the risk of leaking uninitialized memory. Consider instead using an explicit initialization, as below:

```cs
Span<ushort> versionNumbers = stackalloc ushort[4] { ushort.MaxValue, ushort.MaxValue, ushort.MaxValue, ushort.MaxValue };
```

Changing the loop conditional to `i < parts.Length`, and dropping the entire `if ((uint)i >= (uint)parts.Length)` block.